### PR TITLE
Add GetObjectInFrame(x: float, y: float, forceAction: bool)

### PR DESCRIPTION
**Pull request by mattdeitke**
Basically exposes the functionality for `getTargetObject(x, y)`, as requested in #483.

Also, I plan to deprecate every single action that uses `x`, `y`, and `objectId`, in favor of people first calling:
```python
query = controller.step("GetObjectInFrame", x=x, y=y, forceAction=True)
if query:
    objectId = query.metadata['actionReturn']
    controller.step('OpenObject', objectId=objectId)
```

Note that I have overwritten `__bool__` so that it points to `event.metadata['lastActionSuccess']`.

### CODE REVIEW

**Comment by winthos** [comment]
Looks good to me. I assume this syntax will be easier to use rather than passing in object targeting information into each action? That's my only concern, which way is easier to read and use.

**Comment by mattdeitke** [comment]
I don't necessarily know if its easier to read, but I still think it improves a few things better:
1. Making actions more modular. Instead of having a single action that does 2 things, break it down into easier pieces.
2. Will remove the need for overloading these actions with many parameters if there is a python api, since python does not support overloading.

File: ai2thor/server.py [hunk 1]
```diff
@@ -122,6 +122,9 @@ def __init__(self, metadata):
 
         self.events = [self]  # Ensure we have a similar API to MultiAgentEvent
 
+    def __bool__(self):
```
Comment (ekolve) [inline ai2thor/server.py:125]: You should add a __bool__ method into MultiAgentEvent that delegates to self._active_event.__bool__

**Review by ekolve — CHANGES_REQUESTED** [review]

**Comment by ekolve** [comment]
lgtm - i'm fine with the Windows changes as well.  I requested one small change WRT to the new __bool__ overload.  

In terms of documentation, you should consider waiting until the next pip release is made since its currently awaiting validation for arm functionality.  As well, the new __bool__ overload should be documented.

File: ai2thor/server.py (hunk 1 above)
Comment (mattdeitke) [reply ai2thor/server.py:128]: added: https://github.com/allenai/ai2thor/pull/623/commits/2cb3ca45f0721b13003ff3f13cdc7c5b7cbc2877

**Review by ekolve — APPROVED** [review]

**Pull request merged**